### PR TITLE
$ReadOnly

### PR DESCRIPTION
**Pull request by radex**
Adds `$ReadOnly` to a lot of new places. This should fix some of the strange variance issues.

Also, fixes `Q.oneOf()` type from `(string | boolean | number)[]` to `string[] | boolean[] | number[]`

### CODE REVIEW

**Comment by radex** [comment]
@kokusGr CR please

File: src/QueryDescription/index.js [hunk 1]
```diff
@@ -23,7 +23,10 @@ import invariant from '../utils/common/invariant'
 
 import { type TableName, type ColumnName, columnName } from '../Schema'
 
+type $RE<Type> = $ReadOnly<$Exact<Type>>
```
Comment (kokusGr) [inline src/QueryDescription/index.js]: We should make it global type and use it everywhere to avoid confusion :)

Comment (radex) [reply src/QueryDescription/index.js]: Agreed, though in a library we can't really make it "global", we'd need to import it anyway from a file. I could start a `src/types.js` file. A lot of types could be moved there too...

Comment (kokusGr) [reply src/QueryDescription/index.js]: 👌 Are you going to do it in this PR?

Comment (radex) [reply src/QueryDescription/index.js]: sure

**Comment by radex** [comment]
@kokusGr 🏓

**Comment by kokusGr** [comment]
LGTM :)

**Pull request merged**